tests/testthat/test-dn_dry.R

```diff
@@ -0,0 +1,52 @@
+testthat::test_that("This would expect the error with creds_set_aws in fn_dry function",{
+  testthat::expect_error(
+    fn_dry(
+      creds_set_aws(key = "OPIKJDI5LPKnjdBKsGES", secret = "BQLdfuKSsJSUsdIE73KL")
+    )
+  )
+})
+
+testthat::test_that("This would expect the equality in fn_dry function",{
+  testthat::expect_equal(
+    fn_dry(
+      creds_set_aws(key = Sys.getenv("AWS_KEY"), secret = Sys.getenv("AWS_SECRET"))
+    ),
+    "creds_set_aws(key = Sys.getenv(\"AWS_KEY\"), secret = Sys.getenv(\"AWS_SECRET\"))"
+  )
+})
+
+testthat::test_that("This would expect the error with creds_set_labkey in fn_dry function",{
+  testthat::expect_error(
+    fn_dry(
+      creds_set_labkey(api_key = "OPIKJDI5LPKnjdBKsGES")
+    )
+  )
+})
+
+testthat::test_that("This would expect the equality in fn_dry function",{
+  testthat::expect_equal(
+    fn_dry(
+      creds_set_labkey(api_key = Sys.getenv("API_KEY"))
+    ),
+    "creds_set_labkey(api_key = Sys.getenv(\"API_KEY\"))"
+  )
+})
+
+
+testthat::test_that("This would test if labkey api credentials from keyring pass in fn_dry function",{
+  testthat::expect_equal(
+    fn_dry(
+      creds_set_labkey(keyring::key_get(service = "API_KEY", keyring="remote"))
+    ),
+    "creds_set_labkey(keyring::key_get(service = \"API_KEY\", keyring = \"remote\"))"
+  )
+})
+
+testthat::test_that("This would test if aws credentials from keyring pass in fn_dry function",{
+  testthat::expect_equal(
+    fn_dry(
+      creds_set_aws(keyring::key_get(service = "AWS_KEY", keyring="remote"))
```

Comment:
This is a valid check of fn_dry but an incorrect use of creds_set_aws because it only provides the key, not the secret.

Reply:
Done. https://github.com/amashadihossein/dpbuild/commit/8e37f9dc671b6d6a971db5be09feb4ea4205baea